Automattic/jetpack
# Improve Extensibility of VideoPress Privacy Setting filter by adding the embedded post id.

Enhances the privacy setting filter by adding the id of the post that embeds the video as an argument. This is useful because we might want to tie authorization mechanisms that require a post_id to work. Examples are 

* https://woocommerce.com/products/woocommerce-memberships/
* https://woocommerce.com/products/woocommerce-subscriptions/
* https://wordpress.org/plugins/sensei-lms/

Fixes #

#### Changes proposed in this Pull Request:

* Adds an `$embedded_post_id` parameter to the `videopress_is_current_user_authed_for_video` filter.
* Delays enqueueing of token bridge script a bit so we have a post id.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






Here is a small mu-plugin to test with...


```php
<?php
function example_override_video_privacy_setting_for_post( $is_user_authed, $guid, $embedded_post_id = 0 ) {
	$your_post_id = null; /* TODO: add your post id here. */
	if ( $your_post_id === $embedded_post_id ) {
		return true;
	}

	return $is_user_authed;
}

function videopress_override_init() {
	add_filter( 'videopress_is_current_user_authed_for_video', 'example_override_video_privacy_setting_for_post', 10, 3 );
}
add_action( 'init', 'videopress_override_init' );

```

* this can be tested on a local dev env, or in any env where you can add an mu-plugin or a bit of php code.
* Set global privacy setting to private.
* Create and publish 2 posts and embed the same video in both. Note post_ids
* Add the php snippet above to your site. Replace the check with your post id.
* Test the post pages on a private window. The video is public on the post with the id you used on the snippet.
